Framework/Core/src/ASoA.cxx

```diff
@@ -11,17 +11,20 @@
 
 #include "Framework/ASoA.h"
 #include "ArrowDebugHelpers.h"
+#include "Framework/RuntimeError.h"
 
 namespace o2::soa
 {
 
-std::shared_ptr<arrow::Table> ArrowHelpers::joinTables(std::vector<std::shared_ptr<arrow::Table>>&& tables)
+std::shared_ptr<arrow::Table> ArrowHelpers::joinTables(std::vector<std::shared_ptr<arrow::Table>>&& tables, std::vector<const char*> names)
 {
   if (tables.size() == 1) {
     return tables[0];
   }
   for (auto i = 0u; i < tables.size() - 1; ++i) {
-    assert(tables[i]->num_rows() == tables[i + 1]->num_rows());
+    if (tables[i]->num_rows() == tables[i + 1]->num_rows()) {
+      throw o2::framework::runtime_error_f("Tables %s and %s have different sizes and cannot be joined!", names[i], names[i + 1]);
```

Comment:
Can we also print the actual sizes of the two tables?

Reply:
Added.